packages/shared/src/mock/utils.ts

```diff
@@ -1,28 +1,25 @@
 import type { NdArray } from 'ndarray';
 import ndarray from 'ndarray';
 
+import type { Dataset, DType, Shape } from '..';
+import { assertDataset } from '..';
 import {
   assertAbsolutePath,
   isGroup,
   hasChildren,
   assertArray,
   assertArrayShape,
-  isDataset,
   assertDefined,
 } from '../guards';
 import type { Entity } from '../models-hdf5';
 import { getChildEntity } from '../utils';
 import { mockMetadata } from './metadata';
 import type { MockDataset } from './models';
 
-function isMockDataset(entity: Entity): entity is MockDataset {
-  return isDataset(entity) && 'value' in entity;
-}
-
-export function assertMockDataset(
-  entity: Entity
-): asserts entity is MockDataset {
-  if (!isMockDataset(entity)) {
+export function assertMockDataset<S extends Shape, T extends DType>(
+  dataset: Dataset<S, T>
```

Comment:
Unrelated change: the guard now accepts a dataset instead of an entity, which means that callers just need to make sure to call `assertDataset` beforehand if they need to (which is not always the case).